src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,7 +1,78 @@
-import React from 'react';
+import { FC, useState, ChangeEvent } from 'react';
 import './FindMovie.scss';
+import classNames from 'classnames';
+import { Movie } from '../../types/Movie';
+import { getMovie } from '../../api';
+import { MovieCard } from '../MovieCard/MovieCard';
+
+type Props = {
+  movies: Movie[]
+  handleMovieSet: (movie: Movie) => void;
+};
+
+export const FindMovie: FC<Props> = ({ handleMovieSet, movies }) => {
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [title, setTitle] = useState('');
+  const [error, setError] = useState(false);
+  const [isSearching, setSearched] = useState(false);
+
+  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    setTitle(event.target.value);
+    setError(false);
+  };
+
+  const handleGetMovie = (event: React.MouseEvent<HTMLButtonElement,
+  MouseEvent>) => {
+    event.preventDefault();
+    setSearched(true);
+
+    return (title && getMovie(title)
+      .then(data => {
+        if ('Error' in data) {
+          setError(true);
+        } else {
+          setMovie({
+            title: data.Title,
+            description: data.Plot,
+            imgUrl: data.Poster,
+            imdbId: data.imdbID,
+            imdbUrl: `https://www.imdb.com/title/${data.imdbID}`,
+          });
+
+          setError(false);
+        }
+      })
+      .finally(() => {
+        setSearched(false);
+      })
+    );
+  };
+
+  const handleMovieAdd = () => {
+    if (!movie) {
+      return;
+    }
+
+    const foundMovie = movies.find(current => current.title === movie.title);
+
+    if (!foundMovie) {
+      handleMovieSet(movie);
+    }
+
+    setMovie(null);
+    setTitle('');
+    setError(false);
+  };
+
+  const handleSubmit = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
```

Comment:
why do we need to prevent default here?

just add `type="submit"` to your button and `onSubmit={handleSubmit}` to your form (and you can remove onClick event from button)